File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/c/aeropuerto.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "lista.h"
#include "avion.h"
#include "aeropuerto.h"

struct Aeropuerto{
    char nombre[20];
    char direccion[20];
    char telefono[15];

    ListaPtr listaAviones;
};

AeropuertoPtr crearAeropuerto(char* nombre, char* direccion, char* telefono){
    AeropuertoPtr a = (AeropuertoPtr) malloc(sizeof(struct Aeropuerto));
    strcpy(a->nombre, nombre);
    strcpy(a->direccion, direccion);
    strcpy(a->telefono, telefono);

    a->listaAviones = crearLista();

    return a;
}

char* getNombre(AeropuertoPtr a){
    return a->nombre;
}
char* getDireccion(AeropuertoPtr a){
    return a->direccion;
}
char* getTelefono(AeropuertoPtr a){
    return a->telefono;
}

void setNombre(AeropuertoPtr a, char* nombre){
    strcpy(a->nombre, nombre);
}
void setDireccion(AeropuertoPtr a, char* direccion){
    strcpy(a->direccion, direccion);
}
void setTelefono(AeropuertoPtr a, char* telefono){
    strcpy(a->telefono, telefono);
}


ListaPtr getListaAviones(AeropuertoPtr a){
    return a->listaAviones;}



void insertarAvion(AeropuertoPtr a, AvionPtr av){
    insertarUltimo(a->listaAviones, av);
}





void mostrarAeropuerto(AeropuertoPtr a){
    printf("\n---------AEROPUERTO---------\n");
    printf("\nNOMBRE: %s -- DIRECCION: %s -- TELEFONO: %s\n", a->nombre, a->direccion, a->telefono);

    mostrarListaGenerica(a->listaAviones, &mostrarAvionDatoPtr);

}

void liberarAeropuerto(AeropuertoPtr a){
    free(a);
}

File: CLASE VII - Listas Void con Struct Anidadas/pre_parcial/golang/aeropuerto.go
package main

import "fmt"

type Aeropuerto struct {
	nombre    string
	direccion string
	telefono  string

	listaAviones *Lista[*Avion]
}

func crearAeropuerto(nombre, direccion, telefono string) *Aeropuerto {
	return &Aeropuerto{
		nombre:      nombre,
		direccion:   direccion,
		telefono:    telefono,
		listaAviones: crearLista[*Avion](),
	}
}

func (a *Aeropuerto) GetNombre() string          { return a.nombre }
func (a *Aeropuerto) GetDireccion() string       { return a.direccion }
func (a *Aeropuerto) GetTelefono() string        { return a.telefono }
func (a *Aeropuerto) SetNombre(nombre string)    { a.nombre = nombre }
func (a *Aeropuerto) SetDireccion(dir string)    { a.direccion = dir }
func (a *Aeropuerto) SetTelefono(tel string)     { a.telefono = tel }
func (a *Aeropuerto) GetListaAviones() *Lista[*Avion] { return a.listaAviones }

func insertarAvion(a *Aeropuerto, av *Avion) {
	insertarUltimo(a.listaAviones, av)
}

func mostrarAeropuerto(a *Aeropuerto) {
	fmt.Printf("\n---------AEROPUERTO---------\n")
	fmt.Printf("\nNOMBRE: %s -- DIRECCION: %s -- TELEFONO: %s\n", a.nombre, a.direccion, a.telefono)
	mostrarListaGenerica(a.listaAviones, func(av *Avion) { mostrarAvionDatoPtr(av) })
}

